include/sdcard/sdcard.h

```diff
@@ -29,15 +29,34 @@
 
 CPP_GUARD_BEGIN
 
+#define SD_EXTRA_STACK 128
+
+FRESULT sd_open( FIL* file, const TCHAR* path, BYTE mode );
+FRESULT sd_close( FIL* file );
+FRESULT sd_write( FIL* file, const void* buf, UINT btr, UINT* br );
+FRESULT sd_read( FIL* file, void* buff, UINT btr, UINT* br );
+FRESULT sd_lseek( FIL* file, DWORD ofs );
+FRESULT sd_sync( FIL* file );
+FRESULT sd_truncate( FIL* file );
+
+TCHAR*  sd_gets( TCHAR* buff, int len, FIL* fp );
+DWORD   sd_size( FIL* file );
+int	sd_puts (const TCHAR* str, FIL* cp);	
+
 bool test_sd(struct Serial *serial, int lines,int doFlush, int quiet);
 void test_sd_interactive(struct Serial *serial, int lines,int doFlush, int quiet);
 void InitFSHardware(void);
-int InitFS();
-int UnmountFS();
+int InitFS(); // Why doesn't this return FRESULT?
+int UnmountFS(); // Why doesn't this return FRESULT?
+// check the sdcard to see if its ready
+// remount the filesystem if necessary.
+bool sdcard_write_ready( const bool no_delay );
+bool sdcard_ready( const bool no_delay );
```

Comment:
The sdcard_write_ready and sdcard_ready are the functions whose usage I'd like a thorough critique of.  They unify the intitialize/re-initialization of the SD card with a single retry timer, so that regardless of which module is attempting to write(or read) it will only attempt to mount every SD_MOUNT_RETRY_TIME_MS (==1000ms).  Currently the retry counter is not implemented.  Appreciate feedback on this.